src/App.js

```diff
@@ -1,61 +1,48 @@
 import './styles/reset.css'
 import './styles/index.css'
+import { useState } from 'react'
+import React from 'react'
 
-import initialStoreItems from './store-items'
+import initialStoreItems from './Store-items'
+import Store from './components/Store.js'
+import Cart from './components/Cart.js'
 
-/*
-Here's what a store item should look like
-{
-  id: '001-beetroot',
-  name: 'beetroot',
-  price: 0.35
-}
+export default function App() {
+  const [store, setStore] = useState(initialStoreItems)
+  const [cart, setCart] = useState([])
 
-What should a cart item look like? 🤔
-*/
+  // making a new array of a cart by adding items
+  const addNewItem = item => {
+    setCart([...cart, item])
+    console.log('hello'.item)
+  }
 
-console.log(initialStoreItems)
+  // updating items
+  const onIncreaseClick = target => {
+    const updatedQuantity = cart.map(item =>
+      item.id === target.id ? { ...item, quantity: item.quantity + 1 } : item
+    )
+    setCart(updatedQuantity)
+  }
 
-export default function App() {
-  // Setup state here...
+  // removing items
+  const onDecreaseClick = target => {
+    if (target.quantity === 1) {
+      const updatedQuantity = cart.filter(item => item.id !== target.id)
+      setCart(updatedQuantity)
+    } else {
+      const updatedQuantity = cart.map(item =>
+        item.id === target.id ? { ...item, quantity: item.quantity - 1 } : item
+      )
+      setCart(updatedQuantity)
+    }
+  }
+  
+
+  // below is a combination about both of the above functions in one
+  function addToCart(item) {
+    cart.some(element => element.id === item.id)
```

Comment:
nice use of .some method here - I used .find but it was basically the same thing. .some just checks if the item exists in the array.